bom/internal/pom.xml

```diff
@@ -0,0 +1,305 @@
+<?xml version="1.0" encoding="utf-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.hibernate.search</groupId>
+        <artifactId>hibernate-search-parent</artifactId>
+        <version>7.0.0-SNAPSHOT</version>
+        <relativePath>../..</relativePath>
+    </parent>
+
+    <artifactId>hibernate-search-internal-bom</artifactId>
+    <name>Hibernate Search - BOM Internal</name>
+    <packaging>pom</packaging>
+
```

Comment:
You need to add some properties to skip publishing this BOM, I think.
We probably have to publish `public` and `public-build`, but not this one.